authentication/auth.go

```diff
@@ -245,12 +214,32 @@ func generateClaimsV2(claimValues map[string][]string, launcherSchema surveys.La
 	surveyMetadata := make(map[string]interface{})
 	data := make(map[string]interface{})
 
-	if schema.SurveyType == "social" || schema.SurveyType == "health" {
+	if isNonBusinessNonDefaultSurvey(schema.SurveyType) {
 		receiptingKeys := []string{"qid"}
 		surveyMetadata["receipting_keys"] = receiptingKeys
 	}
 
-	getSurveyMetadataFromClaims(claimValues, data, claims, surveyMetadata)
+	/*
```

Comment:
I'm slightly confused about the purpose of this block, as when I remove it the behaviour seems to be the same, and I don't get any duplicates? Is there are specific scenario this was added for?

Reply:
It was added it because I thought that `ru_ref` wasn't being handled and would have been placed in the incorrect level for both quick launch and by the UI, seems like it was already being handled else where and should be fine to remove. Would you mind verifying if ru_ref appears in the correct level in the token?